apps/examples/fs_helper/procfs_test.c

```diff
@@ -0,0 +1,161 @@
+#define BUFFER_SIZE 1024
+/****************************************************************************
+ * Private data
+ ****************************************************************************/
+struct pthread_arg {
+	int argc;
+	char **argv;
+};
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+static int read_file(const char *srcpath)
+{
+	int nbytesread;
+	int rdfd;
+	static char buffer[BUFFER_SIZE];
+
+	/* Open the source file for reading */
+
+	rdfd = open(srcpath, O_RDONLY);
+	if (rdfd < 0) {
+		printf("ERROR: Failed to open %s for reading: %s\n", srcpath, strerror(errno));
+		return -1;
+	}
+
+	/* read the file */
+
+	for (;;) {
+		do {
+			nbytesread = read(rdfd, buffer, BUFFER_SIZE);
+			if (nbytesread == 0) {
+				/* End of file */
+
+				close(rdfd);
+				return OK;
+			} else if (nbytesread < 0) {
+				/* EINTR is not an error (but will still stop the copy) */
+
+				printf("ERROR: Read failure: %s\n", strerror(errno));
+				return -1;
+			}
+			printf("read file string[%s]", buffer);
+
+		} while (nbytesread <= 0);
+	}
+	close(rdfd);
+	return OK;
+}
+
+int procfs_commands(void *args)
+{
+	char filename[100] = "/proc/0/status";
+	int ret;
+//  ret = mount(NULL, "/proc", "procfs", 0, NULL);
+//  if(ret < 0){
+//    printf("procfs mount error[%d]\n",ret);
+//    return -1;
+// }
+
+	ret = read_file(filename);
+	if (ret < 0) {
+		printf("procfs error!\n");
+	} else {
+		printf("success procfs test!\n");
+	}
+	if (umount("/proc") < 0) {
+		printf("umount procfs fail\n");
+		return -1;
+	}
+	return 0;
+}
+
+/****************************************************************************
+ * fs_command_main
+ ****************************************************************************/
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int procfs_test_main(int argc, char **argv)
+#endif
+{
+	pthread_t tid;
+	pthread_attr_t attr;
+	struct sched_param sparam;
+	int r;
+
+	struct pthread_arg args;
+
+	args.argc = argc;
+	args.argv = argv;
+
+	/* Initialize the attribute variable */
+	r = pthread_attr_init(&attr);
+	if (r != 0) {
+		printf("%s: pthread_attr_init failed, status=%d\n", __func__, r);
+	}
+
+	/* 1. set a priority */
+	sparam.sched_priority = 100;
+	r = pthread_attr_setschedparam(&attr, &sparam);
+	if (r != 0) {
+		printf("%s: pthread_attr_setschedparam failed, status=%d\n", __func__, r);
+	}
+
+	r = pthread_attr_setschedpolicy(&attr, SCHED_RR);
+	if (r != 0) {
+		printf("%s: pthread_attr_setschedpolicy failed, status=%d\n", __func__, r);
+	}
+
+	/* 2. set a stacksize */
+	r = pthread_attr_setstacksize(&attr, 51200);
+	if (r != 0) {
+		printf("%s: pthread_attr_setstacksize failed, status=%d\n", __func__, r);
+	}
+
+	/* 3. create pthread with entry function */
+	r = pthread_create(&tid, &attr, (pthread_startroutine_t) procfs_commands, (void *)&args);
```

Comment:
remove space between (pthread_startroutine_t) and procfs_commands for coding rule.